internal/ir/xds.go

```diff
@@ -638,8 +647,14 @@ type HeaderSettings struct {
 	// PreserveXRequestID configures whether Envoy will keep the x-request-id header if passed for a request that is edge
 	// (Edge request is the request from external clients to front Envoy) and not reset it, which is the current Envoy behaviour.
 	// It defaults to false.
+	// Deprecated: use RequestID instead
 	PreserveXRequestID bool `json:"preserveXRequestID,omitempty" yaml:"preserveXRequestID,omitempty"`
```

Comment:
Can we just delete this field and use only `RequestID` internally? Do we usually avoid breaking changes at IR level as well?

Reply:
Yah +1 to cleaning this up, can also be done in a follow up